app/server/event_handlers/aws.py

```diff
@@ -249,17 +208,26 @@ def format_cloudwatch_alarm(msg):
 
 def format_api_key_detected(payload, client):
     msg = payload.Message
-    regex = r"API Key with value token='(\w.+)' has been detected in url='(\w.+)'"
+    regex = r"API Key with value token='(\w.+)', type='(\w.+)' and source='(\w.+)' has been detected in url='(\w.+)'!"
     # extract the api key and the github repo from the message
     api_key = re.search(regex, msg).groups()[0]
-    github_repo = re.search(regex, msg).groups()[1]
+    type = re.search(regex, msg).groups()[1]
+    source = re.search(regex, msg).groups()[2]
+    github_repo = re.search(regex, msg).groups()[3]
 
     # We don't want to send the actual api-key through Slack, but we do want the name to be given,
     # so therefore extract the api key name by following the format of a Notify api key
     api_key_name = api_key[7 : len(api_key) - 74]
```

Comment:
If we could extract the service ID here and include it as part of the message, it would save the OL a step!  Here is a regex with some named capture groups you can use:
`(?P<prefix>gcntfy-)(?P<keyname>.*)(?P<service_id>[-A-Za-z0-9]{36})-(?P<key_id>[-A-Za-z0-9]{36})`